pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -337,3 +325,20 @@ func (f *Fit) Score(ctx context.Context, state *framework.CycleState, pod *v1.Po
 
 	return f.score(pod, nodeInfo)
 }
+
+// InsufficientResource describes what kind of resource limit is hit and caused the pod to not fit the node.
+type InsufficientResource struct {
+	ResourceName v1.ResourceName
+	// We explicitly have a parameter for reason to avoid formatting a message on the fly
+	// for common resources, which is expensive for cluster autoscaler simulations.
+	Reason    string
+	Requested int64
+	Used      int64
+	Capacity  int64
+}
+
+// NonResource describes a kind of non-resource reason that causes the pod to not fit the node.
+type NonResource struct {
```

Comment:
Since we can't get rid of the framework dependency at this point, I think we should be using `framework.Status` instead of adding a new struct.

Reply:
That sounds weird to depend on `framework.Status` and `noderesoruces.InefficientResouce` - from two irrelevant packages. I think introducing this temporary struct here makes the semantics more clear.

Reply:
I think `framework.Status` doesn't look clear enough, and the temporary structure is a good way. Furthermore, `noderesoruces.InefficientResource` may be more convenient for migration to `component-helpers`.

Reply:
I tried to make some changes and thought that although the code looks smoother, the logic of `AdmissionResult` is not as clear as the combination of `InsufficientResource` and `NoResource`. I tried to explain it in the remarks.